### PR TITLE
*: move update store to another place

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Ref #5648.

### What is changed and how does it work?
After changing the update interval to 1ms and leader-schedule-limit to 2000:
Screen Shot 2022-11-07 at 11 55 23 AM
Screen Shot 2022-11-07 at 11 55 35 AM
Screen Shot 2022-11-07 at 11 55 46 AM
Screen Shot 2022-11-07 at 11 52 52 AM

After changing the interval to 5ms:
Screen Shot 2022-11-15 at 4 01 43 PM

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -435,6 +436,34 @@ func (c *RaftCluster) runStatsBackgroundJobs() {
...
+			log.Info("update store stats background jobs has been stopped")
+			return
+		case <-ticker.C:
+			// Update related stores.
+			stores := c.GetStores()
+			for _, store := range stores {
+				if store.IsRemoved() {
+					continue
+				}
+				c.core.UpdateStoreStatus(store.GetID())
```
Comment (nolouch) [inline server/cluster/cluster.go:461]: will it affect the balance behavior? such as  balance leader is very sensitive.

**Review by CabinfeverB — COMMENTED** [review]
Maybe need a balance behavior test. Rest LGTM

**Review by nolouch — APPROVED** [review]
lgtm

File: server/cluster/cluster.go [hunk 2]
```diff
@@ -443,6 +445,31 @@ func (c *RaftCluster) runStatsBackgroundJobs() {
...
+
+	ticker := time.NewTicker(updateStoreStatsInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-c.ctx.Done():
+			log.Info("update store stats background jobs has been stopped")
+			return
+		case <-ticker.C:
```
Comment (bufferflies) [inline server/cluster/cluster.go:460]: maybe add some metrics to show the executing?

**Review by bufferflies — APPROVED** [review]

**Pull request merged**